### ISSUE TITLE
用tf2.5 gpu版本跑tf2分支，执行ssq4all_test_v4.py的时候加载模型之后就跳出来 是什么原因

### ISSUE THREAD

**Issue by xiaotangujin** [Liang-yc/ssq#21]
安装版本tf_nightly_gpu-2.5.0. ,cuda 11.1,第一部执行了ssq4all_test_v4.py进行训练，训练到十万的时候手动中断，执行ssq4all_test_v4.py，但是在## loading model from ./model4all_v4/ 之后回到界面
中间自行更改了os.environ["CUDA_VISIBLE_DEVICES"] = "0"跟model_dir = './model4all_v4/'
请教大佬是啥原因

**Comment by Liang-yc** [comment]
> 安装版本tf_nightly_gpu-2.5.0. ,cuda 11.1,第一部执行了ssq4all_test_v4.py进行训练，训练到十万的时候手动中断，执行ssq4all_test_v4.py，但是在## loading model from ./model4all_v4/ 之后回到界面
> 中间自行更改了os.environ["CUDA_VISIBLE_DEVICES"] = "0"跟model_dir = './model4all_v4/'
> 请教大佬是啥原因

盲猜是你改了路径，但是做测试的时候是往`./model4all_v4/`这个默认路径去找模型的。见[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py)line 61

`        checkpoint = tf.train.latest_checkpoint('./model4all_v4/')`

我没设置config，所以测试的时候不会按照训练文件的路径做出调整。
然后提issue最好把错误信息贴出来，方便纠错。

**Comment by xiaotangujin** [comment]
> > 安装版本tf_nightly_gpu-2.5.0. ,cuda 11.1,第一部执行了ssq4all_test_v4.py进行训练，训练到十万的时候手动中断，执行ssq4all_test_v4.py，但是在## loading model from ./model4all_v4/ 之后回到界面
> > 中间自行更改了os.environ["CUDA_VISIBLE_DEVICES"] = "0"跟model_dir = './model4all_v4/'
> > 请教大佬是啥原因
> 
> 盲猜是你改了路径，但是做测试的时候是往`./model4all_v4/`这个默认路径去找模型的。见[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py) line 65
> 
> ` checkpoint = tf.train.latest_checkpoint('./model4all_v4/')`
> 
> 我没设置config，所以测试的时候不会按照训练文件的路径做出调整。
> 然后提issue最好把错误信息贴出来，方便纠错。

![image](https://user-images.githubusercontent.com/77447532/105038060-d1b92500-5a99-11eb-854b-d8a0b9b8bafc.png)
![image](https://user-images.githubusercontent.com/77447532/105038073-d5e54280-5a99-11eb-8ba3-afec9f1813f7.png)
信息如上，这个路径已经也添加进去，但是就是在提示之后没反应，我是应该，此文件夹也有文件产生

**Comment by Liang-yc** [comment]
> > > 安装版本tf_nightly_gpu-2.5.0. ,cuda 11.1,第一部执行了ssq4all_test_v4.py进行训练，训练到十万的时候手动中断，执行ssq4all_test_v4.py，但是在## loading model from ./model4all_v4/ 之后回到界面
> > > 中间自行更改了os.environ["CUDA_VISIBLE_DEVICES"] = "0"跟model_dir = './model4all_v4/'
> > > 请教大佬是啥原因
> > 
> > 
> > 盲猜是你改了路径，但是做测试的时候是往`./model4all_v4/`这个默认路径去找模型的。见[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py) line 65
> > ` checkpoint = tf.train.latest_checkpoint('./model4all_v4/')`
> > 我没设置config，所以测试的时候不会按照训练文件的路径做出调整。
> > 然后提issue最好把错误信息贴出来，方便纠错。
> 
> ![image](https://user-images.githubusercontent.com/77447532/105041618-2c548000-5a9e-11eb-9f82-7155b491a54b.png)
> ![image](https://user-images.githubusercontent.com/77447532/105038073-d5e54280-5a99-11eb-8ba3-afec9f1813f7.png)
> 信息如上，这个路径已经也添加进去，但是就是在提示之后没反应，我是应该，此文件夹也有文件产生

从你的输出来看，建议如下操作：
1.改用CPU运行，看能否正常出结果。现在`tf2`分支下[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py)加了以下代码，是用CPU出结果的。
`import os
os.environ["CUDA_VISIBLE_DEVICES"]="-1"  
`
然后需要注意的是我改了模型结构`ssq4all_test_v4.py` line 52，`vocab_size=35+12,`改成了`vocab_size=33+16,`,你需要根据你的`ssq4all_v4.py`的网络结构调成正确的结构。

2.用pycharm，python idle等打开工程，运行`ssq4all_test_v4.py`，应该能显示更多错误信息。

另外，你手动中断的时候，是否明确看到了`## Last epoch were saved, next time will start from epoch`这样的输出？如果没有，说明模型没有完整保存好。

**Comment by xiaotangujin** [comment]
> 现在`tf2`分支下[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py)改了line 47-line49代码，我这边测试通过了。你的代码改完后应该也能直接过@xiaotangujin。tf1.x升级至tf2.x时我只测试了训练代码，没测试test部分的代码（tf 2.x似乎不再支持维度大小为0的tensor），抱歉。
> 
> ```
>     input_data = tf.placeholder(tf.float32, [1, 10,7+8,1])
>     logits = inference(input_data, 1, reuse=False,output_num=128)
> 
>     # print(tf.shape(input_data))
>     output_targets = tf.placeholder(tf.int32, [1, None])
>     end_points = rnn_model(model='lstm', input_data=logits, output_data=output_targets, vocab_size=33+16,output_num=7,
>                            rnn_size=128, num_layers=7, batch_size=1, learning_rate=0.001)
> ```

大佬 最近又试了一下tf2.0的几个，发现双色球两个版本都能能训练，但是进行预测的时候就会出错，
logits = inference(input_data, 1, reuse=False,output_num=128)，貌似执行这个的时候出现问题，用idle运行也没弹出出错信息，反而是大乐透能预测出，但是第六个值会大于35，第七个值反而没问题

**Comment by Liang-yc** [comment]
> > 现在`tf2`分支下[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py)改了line 47-line49代码，我这边测试通过了。你的代码改完后应该也能直接过@xiaotangujin。tf1.x升级至tf2.x时我只测试了训练代码，没测试test部分的代码（tf 2.x似乎不再支持维度大小为0的tensor），抱歉。
> > ```
> >     input_data = tf.placeholder(tf.float32, [1, 10,7+8,1])
> >     logits = inference(input_data, 1, reuse=False,output_num=128)
> > 
> >     # print(tf.shape(input_data))
> >     output_targets = tf.placeholder(tf.int32, [1, None])
> >     end_points = rnn_model(model='lstm', input_data=logits, output_data=output_targets, vocab_size=33+16,output_num=7,
> >                            rnn_size=128, num_layers=7, batch_size=1, learning_rate=0.001)
> > ```
> 
> 大佬 最近又试了一下tf2.0的几个，发现双色球两个版本都能能训练，但是进行预测的时候就会出错，
> logits = inference(input_data, 1, reuse=False,output_num=128)，貌似执行这个的时候出现问题，用idle运行也没弹出出错信息，反而是大乐透能预测出，但是第六个值会大于35，第七个值反而没问题

你找我说的把ssq4all_test_v4.py的line 47-line49代码改了，双色球测试应该也就正常了，你重新下`tf2`分支好了。大乐透第6个数字大于35是因为没有-34.

**Comment by xiaotangujin** [comment]
> > > 现在`tf2`分支下[ssq4all_test_v4.py](https://github.com/Liang-yc/ssq/blob/tf2/ssq4all_test_v4.py)改了line 47-line49代码，我这边测试通过了。你的代码改完后应该也能直接过@xiaotangujin。tf1.x升级至tf2.x时我只测试了训练代码，没测试test部分的代码（tf 2.x似乎不再支持维度大小为0的tensor），抱歉。
> > > ```
> > >     input_data = tf.placeholder(tf.float32, [1, 10,7+8,1])
> > >     logits = inference(input_data, 1, reuse=False,output_num=128)
> > > 
> > >     # print(tf.shape(input_data))
> > >     output_targets = tf.placeholder(tf.int32, [1, None])
> > >     end_points = rnn_model(model='lstm', input_data=logits, output_data=output_targets, vocab_size=33+16,output_num=7,
> > >                            rnn_size=128, num_layers=7, batch_size=1, learning_rate=0.001)
> > > ```
> > 
> > 
> > 大佬 最近又试了一下tf2.0的几个，发现双色球两个版本都能能训练，但是进行预测的时候就会出错，
> > logits = inference(input_data, 1, reuse=False,output_num=128)，貌似执行这个的时候出现问题，用idle运行也没弹出出错信息，反而是大乐透能预测出，但是第六个值会大于35，第七个值反而没问题
> 
> 你找我说的把ssq4all_test_v4.py的line 47-line49代码改了，双色球测试应该也就正常了，你重新下`tf2`分支好了。大乐透第6个数字大于35是因为没有-34.

今天试了，可以了，大佬厉害
